legacy-fabric-logger-api-v1/common/src/main/java/net/legacyfabric/fabric/impl/logger/utils/TriState.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2020 - 2024 Legacy Fabric
+ * Copyright (c) 2016 - 2022 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.legacyfabric.fabric.impl.logger.utils;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BooleanSupplier;
+import java.util.function.Supplier;
+
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Represents a boolean value which can be true, false or refer to a default value.
+ */
+public enum TriState {
```

Comment:
wait why does this exist

Reply:
https://github.com/Legacy-Fabric/fabric/blob/main/legacy-fabric-api-base/common/src/main/java/net/legacyfabric/fabric/api/util/TriState.java exists already. I think sponge command api uses it

Reply:
logger-api was supposed to be a standalone module, and technically every module depends on it so... what should we do in this case?

Reply:
> logger-api was supposed to be a standalone module, and technically every module depends on it so... what should we do in this case?

that's fine, logger api could depend on api base. could also move logger api into api base

Reply:
Hm, I know some projects that uses it outside of LegacyFabric too...